h5netcdf/tests/test_h5netcdf.py

```diff
@@ -158,14 +162,29 @@ def read_legacy_netcdf(tmp_netcdf, read_module, write_module):
     assert v.chunking() == 'contiguous'
     assert v.filters() == {'complevel': 0, 'fletcher32': False,
                            'shuffle': False, 'zlib': False}
+    ds.close()
 
-    v = ds.variables['z']
-    assert array_equal(v, _char_array)
-    assert v.dtype == 'S1'
-    assert v.ndim == 2
-    assert v.dimensions == ('z', 'string3')
-    assert v.ncattrs() == ['_FillValue']
-    assert v.getncattr('_FillValue') == b'X'
+    #Check the behavior if h5py. Cannot expect h5netcdf to overcome these errors:
+    with h5py.File(tmp_netcdf, 'r') as ds:
```

Comment:
Can you move this to a separate utility function, so we can reuse it in both tests? Also please note the underlying bug in netCDF-C in a comment:
https://github.com/Unidata/netcdf-c/issues/298